src/main/java/org/spongepowered/common/config/type/GlobalConfig.java

```diff
@@ -58,46 +58,51 @@
     @Setting(comment = "Configuration options related to the Sql service, including connection aliases etc")
     private SqlCategory sql = new SqlCategory();
 
-    @Setting
+    @Setting(comment = "Configuration options related to commands, including command aliases and hidden commands.")
     private CommandsCategory commands = new CommandsCategory();
 
-    @Setting
+    @Setting(comment = "Configuration options related to permissions and permission handling")
     private PermissionCategory permission = new PermissionCategory();
 
-    @Setting(value = "modules")
+    @Setting(value = "modules", comment = "Options related to optional modules, that can be enabled or disabled.")
     private ModuleCategory mixins = new ModuleCategory();
 
-    @Setting("ip-sets")
+    @Setting(value = "ip-sets", comment = ""
+            + "Automatically assigns (permission) contexts to users that use any of the given ips.\n"
+            + "This can be used to restrict/grand permissions, based on the player's source or target ip.")
     private Map<String, List<IpSet>> ipSets = new HashMap<>();
 
-    @Setting(value = "bungeecord")
+    @Setting(value = "bungeecord", comment = "Configuration options related for server proxies, such as bungeecord")
```

Comment:
`Controls how Sponge interacts with server proxies, such as BungeeCord and Velocity. Requires that the 'bungeecord' module is enabled.`

Reply:
Changed